docs/src/dev-docs/debugging.md

```diff
@@ -0,0 +1,86 @@
+# Investigating shotover bugs
+
+This document describes the general process I follow for debugging.
+There is no need to strictly adhere to it, but maybe you will find something useful here.
+
+## Failing test
+
+Write or modify an integration test to reproduce the bug.
+Integration tests live at `shotover-proxy/tests/`.
+
+## Add DebugPrinter
+
+Add `DebugPrinter` to the test's `topology.yaml`.
+This will setup shotover to log every request and response that passes through.
+For example:
+
+```yaml
+sources:
+  - Kafka:
+      name: "kafka"
+      listen_addr: "127.0.0.1:9192"
+      chain:
+        - DebugPrinter # Add this line here!
+        - KafkaSinkSingle:
+            destination_port: 9092
+            connect_timeout_ms: 3000
+```
+
+For a simple redis request/response, the logs will look like:
+
+```plain
+shotover   06:37:14.712042Z  INFO connection{id=2 source="redis"}: shotover::transforms::debug::printer: Request: Redis Array([BulkString(b"GET"), BulkString(b"bar")])
+shotover   06:37:14.712212Z  INFO connection{id=2 source="redis"}: shotover::transforms::debug::printer: Response: Redis BulkString(b"foo")
+```
+
+## Run the test
+
+Run the test by:
+
+```shell
+cargo nextest run --no-default-features --features kafka kafka_int_tests::passthrough_standard --nocapture
+```
+
+Broken down we have:
+
+* `--no-default-features --features kafka` will compile only the parts of shotover and the integration tests required for the specified protocol.
+  * This will drastically improve time to compile shotover and the tests, currently it drops from 15s to 5s. The difference will grow as shotover grows.
+  * For a full list of features refer to the `[features]` section in [shotover-proxy/Cargo.toml](https://github.com/shotover/shotover-proxy/blob/main/shotover-proxy/Cargo.toml)
+* `kafka_int_tests::passthrough_standard::case_1` the exact name of the test to run.
```

Comment:
Doesn't match the shell command above. While you explain in the line below how to filter, I think this explanation should match the shell command.

Reply:
good catch!